osquery/events/windows/etw/etw_publisher_processes.cpp

```diff
@@ -0,0 +1,552 @@
+      path.replace(pos, hardVolume.length(), logicalDrive);
+      break;
+    }
+  }
+}
+
+void EtwPublisherProcesses::updateUserInfo(const std::string& userSid,
+                                           std::string& username) {
+  // Updating user information using gathered user SIDs as input
+  auto usernameIt = usernamesBySIDs_.find(userSid);
+  if (usernameIt != usernamesBySIDs_.end()) {
+    auto cachedUsername = usernameIt->second;
+    username.assign(cachedUsername);
+  } else {
+    PSID pSid = nullptr;
+
+    if (!ConvertStringSidToSidA(userSid.c_str(), &pSid) || pSid == nullptr) {
+      // Inserting empty username to avoid the lookup logic to be called again
+      usernamesBySIDs_.insert({userSid, ""});
+      return;
+    }
+
+    std::vector<char> domainNameStr(MAX_PATH - 1, 0x0);
+    std::vector<char> userNameStr(MAX_PATH - 1, 0x0);
+    DWORD domainNameSize = MAX_PATH;
+    DWORD userNameSize = MAX_PATH;
+    SID_NAME_USE sidType = SID_NAME_USE::SidTypeInvalid;
+
+    if (!LookupAccountSidA(NULL,
+                           pSid,
+                           userNameStr.data(),
+                           &userNameSize,
+                           domainNameStr.data(),
+                           &domainNameSize,
+                           &sidType) ||
+        strlen(domainNameStr.data()) == 0 ||
+        strlen(domainNameStr.data()) >= MAX_PATH ||
+        strlen(userNameStr.data()) == 0 ||
+        strlen(userNameStr.data()) >= MAX_PATH ||
+        sidType == SID_NAME_USE::SidTypeInvalid) {
+      // Inserting empty username to avoid the lookup logic to be called again
+      LocalFree(pSid);
+      return;
+    }
+
+    LocalFree(pSid);
+
+    username.append(domainNameStr.data());
+    username.append("\\");
+    username.append(userNameStr.data());
+
+    usernamesBySIDs_.insert({userSid, username});
+  }
+}
+
+void EtwPublisherProcesses::updateImagePath(const std::uint64_t& key1,
+                                            const std::uint64_t& key2,
+                                            std::string& imagePath) {
+  // This is the search key to be used on the cache of process start events
+  std::uint64_t searchKey = getComposedKey(key1, key2);
+
+  // Event specific post processing callback logic
+  imagePath = tryTake(processImageCache_, searchKey).takeOr(imagePath);
+}
+
+void EtwPublisherProcesses::updateTokenInfo(const std::uint32_t& tokenType,
+                                            std::string& tokenInfo) {
+  // Updating token information with descriptive type
+  switch (tokenType) {
+  case TOKEN_ELEVATION_TYPE::TokenElevationTypeDefault: {
+    tokenInfo.assign("TokenElevationTypeDefault");
+  } break;
+
+  case TOKEN_ELEVATION_TYPE::TokenElevationTypeFull: {
+    tokenInfo.assign("TokenElevationTypeFull");
+  } break;
+
+  case TOKEN_ELEVATION_TYPE::TokenElevationTypeLimited: {
+    tokenInfo.assign("TokenElevationTypeLimited");
+  } break;
```

Comment:
Let's use the same `break` in the curly braces syntax here please.

Reply:
Fixed